Orange/data/table/base.py

```diff
@@ -0,0 +1,1271 @@
+import os
+import zlib
+from collections import Sequence
+from threading import Lock
+from numbers import Number
+
+from itertools import chain
+import numpy as np
+import scipy.sparse as sp
+import pandas as pd
+import pandas.core.internals
+
+from Orange.data import Domain, StringVariable, ContinuousVariable, DiscreteVariable, Variable, TimeVariable
+from Orange.util import flatten, deprecated
+
+
+# noinspection PyPep8Naming
+class TableBase:
+    """An abstract base class for data storage structures in Orange."""
+    KNOWN_PANDAS_KWARGS = {}
+
+    # the default name for weights columns
+    # subclasses may override this to rename the column
+    _WEIGHTS_COLUMN = "__weights__"
+
+    # a collection of all known internal column names
+    # use case: Corpus adds additional columns and we need to know
+    # of all available names
```

Comment:
Egg when concatenating columns, we skip the internals because they would otherwise be duplicated.